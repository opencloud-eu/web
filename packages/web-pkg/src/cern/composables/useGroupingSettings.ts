import { IncomingShareResource } from '@opencloud-eu/web-client'
import { computed, Ref, unref } from 'vue'

export const useGroupingSettings = ({
  sortBy,
  sortDir
}: {
  sortBy: Ref<string>
  sortDir: Ref<string>
}) => {
  const groupingSettings = computed(() => {
    return {
      groupingBy: localStorage.getItem('grouping-shared-with-me') || 'Shared on',
      showGroupingOptions: true,
      groupingFunctions: {
        'Name alphabetically': function (row: IncomingShareResource) {
          localStorage.setItem('grouping-shared-with-me', 'Name alphabetically')
          if (!isNaN(Number(row.name.charAt(0)))) {
            return '#'
          }
          if (row.name.charAt(0) === '.') {
            return row.name.charAt(1).toLowerCase()
          }
          return row.name.charAt(0).toLowerCase()
        },
        'Shared on': function (row: IncomingShareResource) {
          localStorage.setItem('grouping-shared-with-me', 'Shared on')
          const recently = Date.now() - 604800000
          const lastMonth = Date.now() - 2592000000
          if (Date.parse(row.sdate) < lastMonth) {
            return 'Older'
          }
          if (Date.parse(row.sdate) >= recently) {
            return 'Recently'
          } else {
            return 'Last month'
          }
        },
        'Share owner': function (row: IncomingShareResource) {
          localStorage.setItem('grouping-shared-with-me', 'Share owner')
          return row?.owner?.displayName
        },
        None: function () {
          localStorage.setItem('grouping-shared-with-me', 'None')
        }
      },
      sortGroups: {
        'Name alphabetically': function (groups: { name: string }[]) {
          // sort in alphabetical order by group name
          const sortedGroups = groups.sort(function (a, b) {
            if (a.name < b.name) {
              return -1
            }
            if (a.name > b.name) {
              return 1
            }
            return 0
          })
          // if sorting is done by name, reverse groups depending on asc/desc
          if (unref(sortBy) === 'name' && unref(sortDir) === 'desc') {
            sortedGroups.reverse()
          }
          return sortedGroups
        },
        'Shared on': function (groups: { name: string }[]) {
          // sort in order: 1-Recently, 2-Last month, 3-Older
          const sortedGroups = []
          const options = ['Recently', 'Last month', 'Older']
          for (const o of options) {
            const found = groups.find((el) => el.name.toLowerCase() === o.toLowerCase())
            if (found) {
              sortedGroups.push(found)
            }
          }
          // if sorting is done by sdate, reverse groups depending on asc/desc
          if (unref(sortBy) === 'sdate' && unref(sortDir) === 'asc') {
            sortedGroups.reverse()
          }
          return sortedGroups
        },
        'Share owner': function (groups: { name: string }[]) {
          // sort in alphabetical order by group name
          const sortedGroups = groups.sort(function (a, b) {
            if (a.name < b.name) {
              return -1
            }
            if (a.name > b.name) {
              return 1
            }
            return 0
          })
          // if sorting is done by owner, reverse groups depending on asc/desc
          if (unref(sortBy) === 'owner' && unref(sortDir) === 'desc') {
            sortedGroups.reverse()
          }
          return sortedGroups
        }
      }
    }
  })

  return {
    groupingSettings
  }
}
