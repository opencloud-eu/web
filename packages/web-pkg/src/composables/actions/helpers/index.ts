export * from './useIsSearchActive'
export * from './useFileActionsDeleteResources'
export * from './useIsFilesAppActive'
export * from './useIsAppActive'
