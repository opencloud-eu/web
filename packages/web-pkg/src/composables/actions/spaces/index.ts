export * from './useSpaceActionsDelete'
export * from './useSpaceActionsDisable'
export * from './useSpaceActionsDuplicate'
export * from './useSpaceActionsEditDescription'
export * from './useSpaceActionsEditQuota'
export * from './useSpaceActionsEditReadmeContent'
export * from './useSpaceActionsRename'
export * from './useSpaceActionsRestore'
export * from './useSpaceActionsShowMembers'
export * from './useSpaceActionsNavigateToTrash'
export * from './useSpaceActionsSetIcon'
export * from './useSpaceActionsDeleteImage'
