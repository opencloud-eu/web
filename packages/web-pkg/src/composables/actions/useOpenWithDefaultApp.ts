import { useFileActions } from './files'
import { Resource, SpaceResource } from '@opencloud-eu/web-client'

export function useOpenWithDefaultApp() {
  const { getDefaultAction } = useFileActions()

  const openWithDefaultApp = ({
    space,
    resource
  }: {
    space: SpaceResource
    resource: Resource
  }) => {
    if (!resource || resource.isFolder) {
      return
    }

    const fileActionsOptions = {
      resources: [resource],
      space: space
    }
    const defaultEditorAction = getDefaultAction({ ...fileActionsOptions, omitSystemActions: true })
    if (defaultEditorAction) {
      defaultEditorAction.handler(fileActionsOptions)
    }
  }

  return { openWithDefaultApp }
}
