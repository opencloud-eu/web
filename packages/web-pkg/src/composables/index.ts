export * from './ability'
export * from './actions'
export * from './appDefaults'
export * from './appProviderService'
export * from './archiverService'
export * from './authContext'
export * from './breadcrumbs'
export * from './clientService'
export * from './clipboard'
export * from './download'
export * from './driveResolver'
export * from './embedMode'
export * from './eventBus'
export * from './fileListHeaderPosition'
export * from './filesList'
export * from './folderLink'
export * from './isTopBarSticky'
export * from './keyboardActions'
export * from './links'
export * from './loadingService'
export * from './localStorage'
export * from './pagination'
export * from './passwordPolicyService'
export * from './piniaStores'
export * from './portalTarget'
export * from './previewService'
export * from './requestHeaders'
export * from './resources'
export * from './router'
export * from './scrollTo'
export * from './search'
export * from './selection'
export * from './service'
export * from './shares'
export * from './sideBar'
export * from './sort'
export * from './spaces'
export * from './upload'
export * from './viewMode'
export * from './webWorkers'
export * from './avatars'
