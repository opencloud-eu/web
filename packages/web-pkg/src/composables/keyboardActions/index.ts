export * from './useKeyboardActions'
export * from './useCropperKeyboardActions'
