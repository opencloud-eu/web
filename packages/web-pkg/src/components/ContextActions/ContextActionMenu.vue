<template>
  <div id="oc-files-context-menu">
    <oc-list
      v-for="(section, sectionIndex) in menuSections"
      :id="`oc-files-context-actions-${section.name}`"
      :key="`section-${section.name}-list`"
      class="oc-files-context-actions"
      :class="getSectionClasses(sectionIndex)"
    >
      <template v-if="section.items">
        <action-menu-item
          v-for="(action, actionIndex) in section.items"
          :key="`section-${section.name}-action-${actionIndex}`"
          :action="action"
          :appearance="appearance"
          :action-options="actionOptions"
          class="context-menu oc-files-context-action oc-rounded oc-menu-item-hover"
        />
      </template>
      <action-menu-drop-item
        v-if="section.drop && (section.drop.items?.length || section.drop.renderOnEmpty)"
        :menu-section-drop="section.drop"
        :appearance="appearance"
        :action-options="actionOptions"
      />
    </oc-list>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'
import ActionMenuItem from './ActionMenuItem.vue'
import { Action, ActionOptions } from '../../composables'
import { AppearanceType } from '@opencloud-eu/design-system/helpers'
import ActionMenuDropItem from './ActionMenuDropItem.vue'

export type MenuSectionDrop = {
  label: string
  name: string
  icon: string
  items?: Action[]
  renderOnEmpty?: boolean
  emptyMessage?: string
}

export type MenuSection = {
  name: string
  items?: Action[]
  drop?: MenuSectionDrop
}

export default defineComponent({
  name: 'ContextActionMenu',
  components: { ActionMenuDropItem, ActionMenuItem },
  props: {
    menuSections: {
      type: Array as PropType<MenuSection[]>,
      required: true
    },
    appearance: {
      type: String as PropType<AppearanceType>,
      default: 'raw'
    },
    actionOptions: {
      type: Object as PropType<ActionOptions>,
      required: true
    }
  },
  methods: {
    getSectionClasses(index: number) {
      const classes: string[] = []
      if (!this.menuSections.length) {
        return classes
      }
      if (index < this.menuSections.length - 1) {
        classes.push('oc-pb-s')
      }
      if (index > 0) {
        classes.push('oc-pt-s')
      }
      if (index < this.menuSections.length - 1) {
        classes.push('oc-files-context-actions-border')
      }
      return classes
    }
  }
})
</script>

<style lang="scss">
.oc-files-context-actions {
  text-align: left;
  white-space: normal;

  > li {
    padding-left: 0 !important;
    padding-right: 0 !important;

    a,
    button,
    span {
      display: inline-flex;
      font-weight: normal !important;
      justify-content: flex-start;
      vertical-align: top;
      width: 100%;
      text-align: left;
    }
  }

  &-border {
    border-bottom: 0.5px solid var(--oc-role-outline-variant);
  }
}
</style>
