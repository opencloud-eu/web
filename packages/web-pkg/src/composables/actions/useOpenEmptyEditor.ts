import { useGettext } from 'vue3-gettext'
import { useGetMatchingSpace } from '../spaces'
import { useAppsStore, useResourcesStore, useSpacesStore } from '../piniaStores'
import { useClientService } from '../clientService'
import { EDITOR_MODE_EDIT, useFileActions } from './files'
import { storeToRefs } from 'pinia'
import { unref } from 'vue'
import { resolveFileNameDuplicate } from '../../helpers'
import { urlJoin } from '@opencloud-eu/web-client'

// open an editor with an empty file within the current folder
export const useOpenEmptyEditor = () => {
  const { getMatchingSpace } = useGetMatchingSpace()
  const spacesStore = useSpacesStore()
  const appsStore = useAppsStore()
  const resourcesStore = useResourcesStore()
  const clientService = useClientService()
  const { $gettext } = useGettext()
  const { openEditor } = useFileActions()
  const { resources, currentFolder } = storeToRefs(resourcesStore)

  const openEmptyEditor = async (appId: string, extension: string) => {
    let destinationSpace = unref(currentFolder) ? getMatchingSpace(unref(currentFolder)) : null
    let destinationFiles = unref(resources)
    let filePath = unref(currentFolder)?.path

    if (!destinationSpace || !unref(currentFolder).canCreate()) {
      destinationSpace = spacesStore.personalSpace
      destinationFiles = (await clientService.webdav.listFiles(destinationSpace)).children
      filePath = ''
    }

    let fileName = $gettext('New file') + `.${extension}`

    if (destinationFiles.some((f) => f.name === fileName)) {
      fileName = resolveFileNameDuplicate(fileName, extension, destinationFiles)
    }

    const emptyResource = await clientService.webdav.putFileContents(destinationSpace, {
      path: urlJoin(filePath, fileName)
    })

    const space = getMatchingSpace(emptyResource)
    const appFileExtension = appsStore.fileExtensions.find(
      ({ app, extension: ext }) => app === appId && ext === extension
    )

    openEditor(appFileExtension, space, emptyResource, EDITOR_MODE_EDIT)
  }

  return {
    openEmptyEditor
  }
}
