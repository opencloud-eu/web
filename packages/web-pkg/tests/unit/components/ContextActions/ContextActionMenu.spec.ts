import ContextActionMenu from '../../../../src/components/ContextActions/ContextActionMenu.vue'
import { Action } from '../../../../src/composables/actions'
import { defaultPlugins, mount } from '@opencloud-eu/web-test-helpers'

describe('ContextActionMenu component', () => {
  it('renders the menu with actions', () => {
    const menuSections = [
      { name: 'section 1', items: [] as Action[] },
      { name: 'section 2', items: [] as Action[] }
    ]
    const { wrapper } = getWrapper(menuSections)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.oc-files-context-actions').exists()).toBeTruthy()
    expect(wrapper.findAll('.oc-files-context-actions').length).toEqual(menuSections.length)
  })

  it('renders the menu with drop menu items', async () => {
    const menuSections = [
      {
        name: 'apps',
        items: [],
        dropItems: [
          {
            label: 'Apps',
            name: 'apps',
            items: [{ label: () => 'Preview' } as Action, { label: () => 'PDF Viewer' } as Action]
          }
        ]
      },
      {
        name: 'actions',
        items: [{ label: () => 'Download' } as Action],
        dropItems: [
          {
            label: 'Actions',
            name: 'actions',
            items: [{ label: () => 'Copy' } as Action, { label: () => 'Paste' } as Action]
          },
          {
            label: 'Delete',
            name: 'delete',
            items: [
              { label: () => 'Delete' } as Action,
              { label: () => 'Delete permanently' } as Action
            ]
          }
        ]
      },
      {
        name: 'sidebar',
        items: [{ label: () => 'Details' } as Action]
      }
    ]
    const { wrapper } = getWrapper(menuSections)
    expect(wrapper.html()).toMatchSnapshot()

    expect(wrapper.findAll('.oc-files-context-actions').length).toEqual(menuSections.length)
    expect(wrapper.findAll('.oc-files-context-action-drop').length).toEqual(3)
  })
})

function getWrapper(menuSections: { name: string; items: Action[] }[]) {
  return {
    wrapper: mount(ContextActionMenu, {
      props: {
        menuSections,
        actionOptions: { resources: [] }
      },
      global: {
        plugins: [...defaultPlugins()]
      }
    })
  }
}
