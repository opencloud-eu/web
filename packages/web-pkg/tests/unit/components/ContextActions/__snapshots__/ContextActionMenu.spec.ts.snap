// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextActionMenu component > renders the menu with actions 1`] = `
"<div id="oc-files-context-menu">
  <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pb-s oc-files-context-actions-border" id="oc-files-context-actions-section 1"> </ul>
  <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pt-s" id="oc-files-context-actions-section 2"> </ul>
</div>"
`;

exports[`ContextActionMenu component > renders the menu with drop menu items 1`] = `
"<div id="oc-files-context-menu">
  <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pb-s oc-files-context-actions-border" id="oc-files-context-actions-apps">
    <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover"><button type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw oc-py-s oc-px-m oc-width-1-1 oc-flex-justify-between oc-width-1-1" id="oc-files-context-actions-apps-toggle-2">
        <!--v-if-->
        <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m"><svg data-testid="inline-svg-stub" src="icons/info-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span> <span class="oc-flex oc-files-context-action-label"><span>Apps</span></span> <span class="oc-icon oc-icon-s"><svg data-testid="inline-svg-stub" src="icons/arrow-right-s-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span>
      </button>
      <div id="oc-files-context-actions-apps-drop-1" class="oc-drop oc-box-shadow-medium oc-rounded oc-width-auto oc-files-context-action-drop">
        <div class="oc-card oc-card-body oc-p-s">
    <li class="oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
        <!--v-if-->
        <!-- @slot Content of the button -->
        <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>Preview</span></span>
        <!--v-if-->
      </button></li>
    <li class="oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
        <!--v-if-->
        <!-- @slot Content of the button -->
        <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>PDF Viewer</span></span>
        <!--v-if-->
      </button></li>
</div>
</div>
</li>
</ul>
<ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pb-s oc-pt-s oc-files-context-actions-border" id="oc-files-context-actions-actions">
  <li class="context-menu oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
      <!--v-if-->
      <!-- @slot Content of the button -->
      <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>Download</span></span>
      <!--v-if-->
    </button></li>
  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover"><button type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw oc-py-s oc-px-m oc-width-1-1 oc-flex-justify-between oc-width-1-1" id="oc-files-context-actions-actions-toggle-4">
      <!--v-if-->
      <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m"><svg data-testid="inline-svg-stub" src="icons/info-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span> <span class="oc-flex oc-files-context-action-label"><span>Actions</span></span> <span class="oc-icon oc-icon-s"><svg data-testid="inline-svg-stub" src="icons/arrow-right-s-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span>
    </button>
    <div id="oc-files-context-actions-actions-drop-3" class="oc-drop oc-box-shadow-medium oc-rounded oc-width-auto oc-files-context-action-drop">
      <div class="oc-card oc-card-body oc-p-s">
  <li class="oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
      <!--v-if-->
      <!-- @slot Content of the button -->
      <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>Copy</span></span>
      <!--v-if-->
    </button></li>
  <li class="oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
      <!--v-if-->
      <!-- @slot Content of the button -->
      <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>Paste</span></span>
      <!--v-if-->
    </button></li>
  </div>
  </div>
  </li>
  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover"><button type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw oc-py-s oc-px-m oc-width-1-1 oc-flex-justify-between oc-width-1-1" id="oc-files-context-actions-delete-toggle-6">
      <!--v-if-->
      <!-- @slot Content of the button --> <span class="oc-icon oc-icon-m"><svg data-testid="inline-svg-stub" src="icons/info-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span> <span class="oc-flex oc-files-context-action-label"><span>Delete</span></span> <span class="oc-icon oc-icon-s"><svg data-testid="inline-svg-stub" src="icons/arrow-right-s-line.svg" transform-source="(svg) => {
      if (__props.accessibleLabel !== &quot;&quot;) {
        const title = document.createElement(&quot;title&quot;);
        title.setAttribute(&quot;id&quot;, svgTitleId.value);
        title.appendChild(document.createTextNode(__props.accessibleLabel));
        svg.insertBefore(title, svg.firstChild);
      };
      return svg;
    }" aria-hidden="true" focusable="false"></svg></span>
    </button>
    <div id="oc-files-context-actions-delete-drop-5" class="oc-drop oc-box-shadow-medium oc-rounded oc-width-auto oc-files-context-action-drop">
      <div class="oc-card oc-card-body oc-p-s">
  <li class="oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
      <!--v-if-->
      <!-- @slot Content of the button -->
      <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>Delete</span></span>
      <!--v-if-->
    </button></li>
  <li class="oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
      <!--v-if-->
      <!-- @slot Content of the button -->
      <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>Delete permanently</span></span>
      <!--v-if-->
    </button></li>
  </div>
  </div>
  </li>
</ul>
<ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pt-s" id="oc-files-context-actions-sidebar">
  <li class="context-menu oc-files-context-action oc-rounded oc-menu-item-hover"><button type="button" aria-label="" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw action-menu-item oc-py-s oc-px-m oc-width-1-1" data-testid="action-handler">
      <!--v-if-->
      <!-- @slot Content of the button -->
      <!--v-if--> <span class="oc-files-context-action-label oc-flex" data-testid="action-label"><span>Details</span></span>
      <!--v-if-->
    </button></li>
</ul>
</div>"
`;
